store/localcachelayer/user_layer.go

```diff
@@ -155,3 +155,21 @@ func (s LocalCacheUserStore) Get(id string) (*model.User, error) {
 	s.rootStore.doStandardAddToCache(s.rootStore.userProfileByIdsCache, id, user)
 	return user, nil
 }
+
+// GetMany is a cache wrapper around the SqlStore method to get a user profiles by ids.
+// It checks if the user entries are present in the cache, returning the entries from cache
+// if it is present. Otherwise, it fetches the entries from the store and stores it in the
+// cache.
+func (s LocalCacheUserStore) GetMany(ids []string) ([]*model.User, error) {
```

Comment:
This is just going back to the previous implementation. This layer is going to be executed always, so... GetMany is not going to be called ever, because here you use the `s.Get` to get the user.

My Expectation is to loop over the ids, try to get them from the cache, and do a `s.rootStore.GetMany` of the `notFoundIds`, then update the cache with that new obtained users, and return the final list.

Reply:
Damn you are right, silly me. Will amend. i wanted to keep only the caching check for the Get and I somehow thought that it will do that.

Reply:
That should do it 🤞 😄 https://github.com/mattermost/mattermost-server/pull/16819/commits/db9d12df918a14eadcc742ed11a43eb9b3688b51